tests/test_revisions.py

```diff
@@ -65,3 +69,46 @@ def test_check_author_planned_changes_changes_planned(phabdouble):
         attachments={"reviewers": True, "reviewers-extra": True, "projects": True},
     )["data"][0]
     assert check_author_planned_changes(revision=revision) is not None
+
+
+def test_secure_api_flag_on_public_revision_is_false(client, phabdouble):
```

Comment:
You should probably update one of the integrated stack tests to make something secure, as that would have caught the issue around assuming security based on a single revision's status.

Reply:
Agreed.  I added an integrated transplant test to ensure the warnings are computed correctly and an integrated stack test to make sure the stack values are correct.